src/D2L.CodeStyle.Analyzers/Immutability/ImmutableDefinitionChecker.cs

```diff
@@ -370,6 +371,11 @@ private enum AssignmentQueryKind {
 			Hopeless
 		}
 
+		internal static readonly ImmutableArray<string> KnownImmutableReturningMethods = ImmutableArray.Create(

```

Comment:
Is there a better place for this? Is this even the right way to do it?

Reply:
Ah yeah, see the note I left about `ImmutabilityContext`.